Libraries/LibWeb/HTML/HTMLMediaElement.cpp

```diff
@@ -886,10 +886,15 @@ WebIDL::ExceptionOr<void> HTMLMediaElement::select_resource()
             HTML::main_thread_event_loop().spin_until(GC::create_function(heap(), [&]() { return ran_media_element_task; }));
         };
 
-        // 1. ⌛ If the src attribute's value is the empty string, then end the synchronous section, and jump down to the failed with attribute step below.
+        // 1. ⌛ If the src attribute's value is the empty string, then end the synchronous section
+        // Instead of calling failed_with_attribute synchronously, queue it as a task to ensure
+        // event handlers have a chance to be registered before the error event fires
```

Comment:
The comment here is a direct quote from the HTML spec. Please don't change it. If you suspect the spec is wrong (which does happen) then open an issue on https://github.com/whatwg/html (or the relevent spec) and then link to it here in a comment. eg,

```suggestion
        // 1. ⌛ If the src attribute's value is the empty string, then end the synchronous section, and jump down to the failed with attribute step below.
        // AD-HOC: Queue it as a task to ensure event handlers have a chance to be registered before the error event fires. See spec issue: ...
```

Maybe ask about it on Discord too, someone other than me probably knows this better. There definitely have been places where the spec has forgotten to queue a task like this.

Reply:
I suspect this is related to step 4 of this algorithm not being implemented and the ⌛

Reply:
If I am reading it right, it would mean this would be running as part of a microtask, which probably has the same effect as the queue of the task here, so my feeling is that we need to be implementing step 4 here